pkg/aws/services/tagging.go

```diff
@@ -80,6 +86,66 @@ func NewDefaultTagging(sess *session.Session, region string) *defaultTagging {
 	return &defaultTagging{ResourceGroupsTaggingAPIAPI: api}
 }
 
+// Use VPC Lattice API instead of the Resource Groups Tagging API
+func NewLatticeTagging(sess *session.Session, acc string, region string, vpcId string) *latticeTagging {
+	api := NewDefaultLattice(sess, acc, region)
+	return &latticeTagging{Lattice: api, vpcId: vpcId}
+}
+
+func (t *latticeTagging) GetTagsForArns(ctx context.Context, arns []string) (map[string]Tags, error) {
+	result := map[string]Tags{}
+
+	for _, arn := range arns {
+		tags, err := t.ListTagsForResourceWithContext(ctx,
+			&vpclattice.ListTagsForResourceInput{ResourceArn: aws.String(arn)},
+		)
+		if err != nil {
+			return nil, err
+		}
+		result[arn] = tags.Tags
+	}
+	return result, nil
+}
+
+func (t *latticeTagging) FindResourcesByTags(ctx context.Context, resourceType ResourceType, tags Tags) ([]string, error) {
+	if resourceType != ResourceTypeTargetGroup {
+		return nil, fmt.Errorf("unsupported resource type %q for FindResourcesByTags", resourceType)
+	}
+
+	tgs, err := t.ListTargetGroupsAsList(ctx, &vpclattice.ListTargetGroupsInput{
+		VpcIdentifier: aws.String(t.vpcId),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	arns := make([]string, 0, len(tgs))
+
+	for _, tg := range tgs {
+		resp, err := t.ListTagsForResourceWithContext(ctx,
+			&vpclattice.ListTagsForResourceInput{ResourceArn: tg.Arn},
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		if containsTags(tags, resp.Tags) {
+			arns = append(arns, aws.StringValue(tg.Arn))
+		}
+	}
+
+	return arns, nil
+}
+
+func containsTags(source, check Tags) bool {
+	for k, v := range source {
+		if aws.StringValue(check[k]) != aws.StringValue(v) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
You might want to create some separate unit tests just for this. I actually think this logic is currently incorrect. For example, I believe a `source` with tag `"X"=""` will return true even if the `check` does not contain the `"X"` tag. I think if you had something like `checkV, ok := check[k]` and ensure `ok` is `true` that would be enough.

Here, you would want to comprehensively check all the options between the two sets where you want it to return true or false. I listed a few above, but you might want more here like tags being present, having different values, or having matching keys but with different tags. 

This logic has to be super precise, so let's be sure to get good edge case coverage here.

Reply:
you are correct about the error!  I created a unit test where this failed and corrected it in the code as you suggested to make the test pass. Note that I was using the arguments backwards and now range over check (the tags we want to see if the source contains).